### PR TITLE
add ability to support all projects

### PR DESCRIPTION
**Pull request by jtrobman**

### CODE REVIEW

**Comment by gsingers** [comment]
Sounds interesting.  Taking a look.

File: lib/bot.js [hunk 1]
```diff
@@ -119,15 +119,20 @@ Bot.prototype.run = function () {
...
   var prjNames = "";
-  _.each(self.config.projects, function (prj, index, list) {
-    pattern += prj;
-    prjNames += prj;
-    if (index != list.length - 1) {
-      pattern += "|";
-      prjNames += "|";
-    }
-  });
-  pattern += ")-[1-9][0-9]*)(\\+)?(?:(?=\\W)|$)";
```
Comment (gsingers) [inline lib/bot.js:130]: https://github.com/gsingers/slack-jira-plugin/pull/30 is the reason we have the leading [1-9] there.

File: lib/bot.js [hunk 2]
```diff
@@ -95,7 +95,7 @@ Bot.prototype.run = function () {
   var self = this,
       verbose = self.config.verbose,
       bot = new slackbot(this.config.token),
-      pattern = "(?:[^a-zA-Z0-9_/-]|^)((";
```
Comment (gsingers) [inline lib/bot.js:98]: https://github.com/gsingers/slack-jira-plugin/pull/38 This pattern is used to reject matching on issues where the full URL is already given.  e.g. if a user puts in https://my.jira.com/FOO-18, we don't then want to link FOO-18 again.

File: lib/bot.js [hunk 3]
```diff
@@ -119,15 +119,20 @@ Bot.prototype.run = function () {
...
-    pattern += prj;
-    prjNames += prj;
-    if (index != list.length - 1) {
-      pattern += "|";
-      prjNames += "|";
-    }
-  });
-  pattern += ")-[1-9][0-9]*)(\\+)?(?:(?=\\W)|$)";
+  if (self.config.projects !== undefined && self.config.projects.length > 0) {
+    pattern = "(?:\\W)((";
```
Comment (gsingers) [inline lib/bot.js:123]: See #38 comment on the old side.

Comment (gsingers) [reply lib/bot.js:123]: I suppose one of these days, I should write unit tests for all of this so it's more clear all the edge cases.  :-/

**Review by gsingers — COMMENTED** [review]
Thanks for the PR.  I like the idea of it, but I worry that it will yield a lot of false positives.
